### PR TITLE
Add missing guide from kuadrant-operator

### PR DESCRIPTION
**Pull request by Boomatang**
This guide was missing form the site.

### CODE REVIEW

File: mkdocs.yml [hunk 1]
```diff
@@ -95,6 +95,7 @@ nav:
       - 'Metrics Walkthrough': multicluster-gateway-controller/docs/how-to/metrics-walkthrough.md
     - 'Authentication & Authorization':
       - 'AuthPolicy for Application Developers and Platform Engineers': kuadrant-operator/doc/user-guides/auth-for-app-devs-and-platform-engineers.md
+      - 'Authenticated Rate Limiting for Application Developers': kuadrant-operator/doc/user-guides/authenticated-rl-for-app-developers.md
```
Comment (guicassolato) [inline mkdocs.yml:98]: I think I'd probably add it under the "Rate Limiting" guides instead, like between 'RateLimitPolicy for Platform Engineers' and the more complicated 'authenticated-rl-with-jwt-and-k8s-authnz.md' one. WDYT?

Comment (Boomatang) [reply mkdocs.yml:98]: Sure ya, that works for me.

File: mkdocs.yml [hunk 2]
```diff
@@ -123,6 +123,7 @@ nav:
       - 'Observability': authorino/docs/user-guides/observability.md
     - 'Rate Limiting':
       - 'RateLimitPolicy for Application Developers': multicluster-gateway-controller/docs/how-to/simple-ratelimitpolicy-for-app-developers.md
+      - 'Authenticated Rate Limiting for Application Developers': kuadrant-operator/doc/user-guides/authenticated-rl-for-app-developers.md
       - 'RateLimitPolicy for Platform Engineers': kuadrant-operator/doc/user-guides/gateway-rl-for-cluster-operators.md
```
Comment (guicassolato) [inline mkdocs.yml:127]: I hate myself. Sorry @Boomatang. Do you mind grouping these 2 together?

There's a continuity between 1st and 2nd on this list, then the 2 others will be about authenticated RL (first a simpler one, then a more complex one.)

```suggestion
      - 'RateLimitPolicy for Platform Engineers': kuadrant-operator/doc/user-guides/gateway-rl-for-cluster-operators.md
      - 'Authenticated Rate Limiting for Application Developers': kuadrant-operator/doc/user-guides/authenticated-rl-for-app-developers.md
```

**Review by guicassolato — APPROVED** [review]

**Pull request merged**